test/powershell/Modules/Microsoft.PowerShell.Utility/XMLCommand.Tests.ps1

```diff
@@ -159,4 +159,97 @@ Describe "XmlCommand DRT basic functionality Tests" -Tags "CI" {
 		$cmd.Xml = $xml
 		$cmd.Xml | Should -Be $xml
 	}
+
+    It "ConvertFrom-CliXml directive should work" {
+        $content = Get-Command export* -Type Cmdlet | Select-Object -First 3 | ConvertTo-Clixml
+		$results = ConvertFrom-CliXml $content
+		$results.Count | Should -Be 3
+        $results[0].PSTypeNames[0] | Should -BeExactly "Deserialized.System.Management.Automation.CmdletInfo"
+    }
+
+    It "ConvertFrom-CliXml with Rehydration should work" {
+		$property1 = 256
+		$property2 = "abcdef"
+		$isHiddenTestType = [IsHiddenTestType]::New($property1,$property2)
+		$content = $isHiddenTestType | ConvertTo-Clixml
+		$results = ConvertFrom-CliXml $content
+		$results.Property1 | Should -Be $property1
+		$results.Property2 | Should -BeExactly $property2
+    }
+
+    It "ConvertTo-Clixml StopProcessing should succeed" {
+        $ps = [PowerShell]::Create()
+        $null = $ps.AddScript("1..10")
+        $null = $ps.AddCommand("foreach-object")
+        $null = $ps.AddParameter("Process", { $_; Start-Sleep -Seconds 1 })
+        $null = $ps.AddCommand("ConvertTo-Clixml")
+
+        Wait-UntilTrue { $ps.BeginInvoke() } -IntervalInMilliseconds 1000
+        $null = $ps.Stop()
+        $ps.InvocationStateInfo.State | Should -BeExactly "Stopped"
+        $ps.Dispose()
+	}
+
+    It "ConvertFrom-CliXml StopProcessing should succeed" {
+		$content = 1,2,3 | ConvertTo-Clixml
+		$ps = [PowerShell]::Create()
+		$ps.AddCommand("Get-Process")
+		$ps.AddCommand("ConvertFrom-CliXml")
+		$ps.AddParameter("InputObject", $content)
+		$ps.BeginInvoke()
+		$ps.Stop()
+		$ps.InvocationStateInfo.State | Should -BeExactly "Stopped"
+	}
+
+    It "ConvertTo-Clixml using -Depth should work" {
+		class Three
+		{
+			[int] $num = 3;
+		}
+
+		class Two
+		{
+			[Three] $three = [Three]::New();
+			[int] $value = 2;
+		}
+
+		class One
+		{
+			[Two] $two = [Two]::New();
+			[int] $value = 1;
+		}
+
+		$one = [One]::New()
+		$content = $one | ConvertTo-Clixml -Depth 2
+		$deserialized_one = ConvertFrom-CliXml -InputObject $content
+		$deserialized_one.Value | Should -Be 1
+		$deserialized_one.two.Value | Should -Be 2
+		$deserialized_one.two.Three | Should -Not -BeNullOrEmpty
+		$deserialized_one.two.three.num | Should -BeNullOrEmpty
+	}
+
+	It "ConvertFrom-CliXml -IncludeTotalCount always returns unknown total count" {
+		# this cmdlets supports paging, but not this switch
+		$content = [PSCustomObject]@{ foo = 1; bar = @{ hello = "world" }} | ConvertTo-Clixml
+		$out = ConvertFrom-CliXml -InputObject $content -IncludeTotalCount
+		$out[0].ToString() | Should -BeExactly "Unknown total count"
+	}
+
+	It "ConvertFrom-CliXml -First and -Skip work together for simple types" {
+		$content = "one","two","three","four" | ConvertTo-Clixml
+		$out = ConvertFrom-CliXml -InputObject $content -First 2 -Skip 1
+		$out.Count | Should -Be 2
```

Comment:
Indentation.

Reply:
@iSazonov I just realized that the indentation issues is due to mix of tab and space. Which one should I use space or tab or is it fine as long as I use it consistently ?

Reply:
Spaces is generally the norm in the repo. The only exceptions I usually see are really old files.